tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,134 @@
+				Eventually(getConditionStatus).WithArguments(namespace, "deployment", deploymentName, "Available").Should(Equal("True"))
+				podName, _ := getPodNameFromLabel(namespace, "control-plane=istio-operator")
+				Eventually(getPodPhase).WithArguments(namespace, podName).Should(Equal("Running"))
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("is applied the istio resource with version %s", version), func() {
+				BeforeAll(func() {
+					Expect(createNamespaceIfNotExists(controlPlaneNamespace)).To(Succeed())
+					Eventually(createIstioCR).WithArguments(version).Should(Succeed())
+				})
+
+				AfterAll(func() {
+					Eventually(deleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed())
+					Eventually(namespaceIsDeleted).WithArguments(controlPlaneNamespace).Should(Equal(true))
+				})
+
+				When("the Istio resource is created", func() {
+					It("updates the Istio resource status to Ready and Running", func() {
+						Eventually(getConditionStatus).WithArguments(controlPlaneNamespace, "istio", istioName, "Reconciled").Should(Equal("True"))
+						Eventually(getConditionStatus).WithArguments(controlPlaneNamespace, "istio", istioName, "Ready").Should(Equal("True"))
+
+						istiodPodName, _ := getPodNameFromLabel(controlPlaneNamespace, "app=istiod")
+						Eventually(getPodPhase).WithArguments(controlPlaneNamespace, istiodPodName).Should(Equal("Running"))
+					})
+
+					Specify("the istio resource version match the applied version", func() {
+						Expect(getInstalledIstioVersion()).Should(ContainSubstring(version))
+					})
+
+					It("istio resource stopped reconciling", func() {
+						Eventually(captureIstiodLast30SecondsOfLog).WithTimeout(120 * time.Second).Should(ContainSubstring("Reconciliation done"))
+					})
+
+					It("only istiod is deployed", func() {
+						deploymentsList, err := getDeployments(controlPlaneNamespace)
+						if err != nil {
+							Fail("Error getting deployments")
+						}
+
+						// If only istiod is deployed, the length of the list should be 1
+						Expect(len(deploymentsList)).To(Equal(1))
+						Expect(deploymentsList[0]).To(Equal("istiod"))
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						if ocp == "true" {
+							Eventually(cniDaemonIsDeployed).WithArguments(namespace).Should(Equal(true))
+
+							podName, err := getPodNameFromLabel(namespace, "k8s-app=istio-cni-node")
+							if err != nil {
+								Fail("Error getting pod name from deployment")
+							}
+							Eventually(getPodPhase).WithArguments(namespace, podName).Should(Equal("Running"))
+
+						} else {
+							Eventually(cniDaemonIsDeployed).WithArguments(namespace).Should(Equal(false))
+						}
+					})
+
+					When("the Istio CR is deleted", func() {
```

Comment:
This is inside `When("the Istio resource is created")`. It should be at the same level.

Reply:
Done